src/VisualStudio/Core/Def/Preview/PreviewUpdater.cs

```diff
@@ -66,17 +72,27 @@ private void UpdateBuffer(TextDocument document)
 
             if (_currentDocumentId != null)
             {
-                Contract.ThrowIfNull(_previewWorkspace, "We shouldn't have a current document if we don't have a workspace.");
-                var currentDocument = _previewWorkspace.CurrentSolution.GetRequiredTextDocument(_currentDocumentId);
+                Contract.ThrowIfNull(previewWorkspace, "We shouldn't have a current document if we don't have a workspace.");
+                var currentDocument = previewWorkspace.Target.CurrentSolution.GetRequiredTextDocument(_currentDocumentId);
                 var currentDocumentText = currentDocument.GetTextSynchronously(CancellationToken.None);
-                _previewWorkspace.CloseDocument(currentDocument, currentDocumentText);
+                previewWorkspace.Target.CloseDocument(currentDocument, currentDocumentText);
+            }
+
+            if (previewWorkspace is null)
+            {
+                var newWorkspace = PreviewDialogWorkspace.Create(document.Project.Solution);
+                if (Interlocked.CompareExchange(ref _previewWorkspace, newWorkspace, null) is not null)
+                    newWorkspace.Dispose();
             }
```

Comment:
i'm def confused about why this is all written this way.  i would presume the PreviewUpdater is for something like the options page.  and deferring the whole workspace stuff just seems totally unnecessary (And makes things so contorted and hard to follow).

Reply:
➡️ Planning to leave this one as-is since it's the same as before